airflow/models/baseoperator.py

```diff
@@ -578,6 +585,25 @@ def __init__(
         if end_date:
             self.end_date = timezone.convert_to_utc(end_date)
 
+        if retries is not None and not isinstance(retries, int):
+            try:
+                parsed_retries = int(retries)
+            except (TypeError, ValueError):
+                raise AirflowException(f"'retries' type must be int, not {type(retries).__name__}")
+            self.log.warning("Implicitly converting 'retries' for %s from %r to int", self, retries)
```

Comment:
```suggestion
            self.log.warning("Implicitly converting 'retries' for task: %s.%s from %r to int", dag.dag_id, task_id, retries)
```

Even if the above mypy problem is not fixed - might be not best idea to log(self) before constructor finishes.

Reply:
Ah yeah, could be.👍🏻